torch/onnx/utils.py

```diff
@@ -267,6 +267,27 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
     return torch_out
 
 
+def _export_module(module, f, operator_export_type=OperatorExportTypes.ONNX, export_type=ExportTypes.PROTOBUF_FILE):
+    # TODO: Don't allocate a in-memory string for the protobuf
+    from torch.onnx.symbolic import _onnx_opset_version
+
+    proto, storage_map = module.export(_onnx_opset_version, operator_export_type)
+
+    if export_type in [ExportTypes.ZIP_ARCHIVE, ExportTypes.COMPRESSED_ZIP_ARCHIVE]:
+        import zipfile
+        compression = zipfile.ZIP_DEFLATED \
+            if export_type == ExportTypes.COMPRESSED_ZIP_ARCHIVE \
+            else zipfile.ZIP_STORED
+        with zipfile.ZipFile(f, 'w', compression=compression) as z:
```

Comment:
when we get @jamesr66a's c++ zip integration, if it comes with producing code, we can consider moving it to c++ to avoid an extra copy

Reply:
Everything should move to C++, it would clean up the API a lot since the exporter could also handle writing entries into the zip file, making the function prototypes simpler (no more export maps, e.g.). This is for James' follow up PR.